### PR TITLE
DOCS: deployable payload flight example

### PR DESCRIPTION
**Pull request by Gui-FernandesBR**
## Pull request type
- [X] Code maintenance (refactoring, formatting, renaming, tests)
- [X] ReadMe, Docs and GitHub maintenance

## Pull request checklist

- Code base maintenance (refactoring, formatting, renaming):

  - [X] Docs have been reviewed and added / updated if needed
  - [X] Lint (`black rocketpy`) has passed locally and any fixes were made
  - [X] All tests (`pytest --runslow`) have passed locally

- Code base additions (for bug fixes / features):

  - [X] Tests for the changes have been added (not necessary here)
  - [X] Docs have been reviewed and added / updated if needed
  - [X] Lint (`black rocketpy`) has passed locally and any fixes were made
  - [X] All tests (`pytest --runslow`) have passed locally

## What is the current behavior?
Nobody knows how to simulate a rocket that presents a deployable payload

## What is the new behavior?
New notebook added to our docs page!
Also, I've made some minor adjustments to ensure all the quaternios stay inside a feasible range (i.e. from -1 up to 1). This was generating problems during some simulations where I could find quaternions being evaluated as 1.000004, for instance, one was breaking the code when trying to 

Anyway the changes are getting into `enh/complex_simulations` branch

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

## Other information
More plots of comparison flights are needed, but they you be treated separated in another PR.
For now, we are already simulating a deployable payload, it just lacks few visualization tools. ( see https://github.com/RocketPy-Team/RocketPy/compare/docs-deployable-payload...enh/complex_simulations_refactor_plots  for more details)

### CODE REVIEW

**Comment by Gui-FernandesBR** [comment]
@MateusStano can you review this one, please? Important part of our milestone!

**Comment by giovaniceotto** [comment]
Here are answers to some TODO's which I have deleted:

TODO: Understand whether we can run Monte Carlo simulations in this case or not. **Ans: We should be able to run Monte Carlo simulations in this case without problems.**
TODO: Should we define a Payload Class?  **Ans: Not for now.**
TODO: Move Everything together from this notebook inside the utilities module? This would make things easier to use, but I have to think. **Ans: Not for now.**

**Review by giovaniceotto — APPROVED** [review]
Great work @Gui-FernandesBR. This notebook presents a quite sophisticated use of initial solutions and compare flights, making a great example for both multistage rockets and deployable payloads.

I have polished the notebook with very few tweaks and also added it to our docs index so that it appears on our docs webpage.

Regarding the modifications to the Flight class, I played around to find the root cause of the grid mismatch error (each Function instance having different source lengths) and sure enough found it to be a simple fix. The old implementation only worked when `tInitial==0`, which may not be the case when an initial solution is given. This has been fixed by changing just two lines of code. Thus, I rolled back some of the changes you hade made.

Regarding quaternions lacking normalization, I agree with you. However, I strongly believed they should be normalized "online" during the numerical integration to keep accuracy, so that is what I did. Thus, I rolled back the method used not normalize the quaternions.

Finally, I had to update a test which had a nonsensical initial solution (all quaternions equal to 0), and also had to update the version of the docs theme.

Go ahead, perform a quick review of my changes, try to run your new notebook from start to finish and if everything goes well, you may merge this.

Let me know if you have any questions. Try reading my commits in order, one by one, to understand the changes I made. They are quite short commits and well ordered.

It was a pleasure working with your code and applying the final touches before the merge. Keep up the amazing work!

File: rocketpy/utilities.py [hunk 1]
```diff
@@ -329,6 +330,26 @@ def compareFlightTrajectories(
...
+        the same order as the trajectories in flight_list
+
+    Returns
+    -------
+    None
+
+    """
+    return None
+
+
```
[the remark this answers is not in the document]
Comment (Gui-FernandesBR) [reply rocketpy/utilities.py:352]: This will be moved to another file, don't worry for now, I'm already working on the improvements regarding this one

**Comment by Gui-FernandesBR** [comment]
Awesome! Thank you for the solid review.

It's amazing how easy it was to understand the code by using the step by step commits, thank you.

Everything is working fine and we can proceed. However, I still don't get how the following magic line:

> # Normalize quaternions
>                     phase.solver.y[6:10] = phase.solver.y[6:10] / sum(phase.solver.y[6:10]**2)

My concern is... Isn't it modyfing, altough slighly, the quaternions values for a regular simulation? By that I mean, 
But anyway, I already checked and the sum of quadratic quaternions over a regular simulation usually do not escape from a 1 +/- 1e-8 range, so at the end you are dividing a number by something reeeeeally close to 1.

Btw the final solution whould looks like: 
image

**Comment by giovaniceotto** [comment]
@Gui-FernandesBR your concern is justified. Since quaternions should remain normalized during the simulation, and this is not always guaranteed by the integration step, several methods have been developed to handle this.

The simplest one would be to just do nothing, which is what we did before. The next option is to normalize after each time step, but this does involve integration errors. The more accurate option would be to use special numerical integration schemes developed specially for quaternions, which guarantee that they remain normalized. Unfortunately, these are very specific and SciPy does not support any of them yet (should we go ahead and add them in the future so that we can make use of them?).

More information can be found in the literature. I leave one interesting article here: https://doi.org/10.2514/1.58558

**Pull request merged**